webapi/src/main/scala/org/knora/webapi/responders/v2/resources/CheckObjectClassConstraints.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2021 - 2024 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.responders.v2.resources
+
+import zio.*
+
+import dsp.errors.*
+import org.knora.webapi.*
+import org.knora.webapi.messages.IriConversions.*
+import org.knora.webapi.messages.*
+import org.knora.webapi.messages.v2.responder.ontologymessages.*
+import org.knora.webapi.messages.v2.responder.resourcemessages.*
+import org.knora.webapi.messages.v2.responder.valuemessages.*
+
+object CheckObjectClassConstraints {
+
+  /**
+   * Checks that values to be created in a new resource are compatible with the object class constraints
+   * of the resource's properties.
+   *
+   * @param values                a map of property IRIs to values to be created (in the internal schema).
+   * @param linkTargetClasses     a map of resources that are link targets to the IRIs of those resource's classes.
+   * @param entityInfo            an [[EntityInfoGetResponseV2]] containing definitions of the classes that all the link targets
+   *                              belong to.
+   * @param clientResourceIDs     a map of IRIs of resources to be created to client IDs for the same resources, if any.
+   * @param resourceIDForErrorMsg something that can be prepended to an error message to specify the client's ID for the
+   *                              resource to be created, if any.
+   */
+  def apply(
+    propertyIri: SmartIri,
+    valuesToCreate: Seq[CreateValueInNewResourceV2],
+    linkTargetClasses: Map[IRI, SmartIri],
+    entityInfo: EntityInfoGetResponseV2,
+    clientResourceIDs: Map[IRI, String],
+    resourceIdForErrorMsg: IRI,
```

Comment:
`propertyIri` and `valuesToCreate` are missing in docstring.

Reply:
Will be fixed in next MR.